packages/recomponents/src/components/r-date-range/r-date-range.vue

```diff
@@ -0,0 +1,312 @@
+                    _.date(selected.start, DateTimeFormats.shortDate),
+                    _.date(selected.end, DateTimeFormats.shortDate),
+                ].join(' - ');
+            },
+        },
+        methods: {
+            getPresetLabel(presetName) {
+                return calendarPresetsLabels[presetName];
+            },
+            getFormattedPresetPeriod(presetName) {
+                const period = this.calendarPresetsPeriods[presetName];
+                if (oneValuePresetsList.includes(presetName)) {
+                    return _.date(period.start, DateTimeFormats.orderDate);
+                }
+                const formatter = value => _.date(value, DateTimeFormats.shortDate);
+                return `${formatter(period.start)} — ${formatter(period.end)}`;
+            },
+            toggle(name) {
+                if (name === 'presets') {
+                    this.$refs.presets.popper.toggle();
+                    this.$refs.calendar.popper.close();
+                } else {
+                    this.$refs.calendar.popper.toggle();
+                    if (this.$refs.presets) {
+                        this.$refs.presets.popper.close();
+                    }
+                }
+            },
+            close() {
+                if (this.$refs.presets) {
+                    this.$refs.presets.popper.close();
+                }
+
+                this.$refs.calendar.popper.close();
+            },
+            relativeFilterChange(presetName) {
+                const period = this.calendarPresetsPeriods[presetName];
+                /**
+                 * The date range selected
+                 * @type {Event}
+                 */
+                this.$emit('input', {
+                    isRelative: true,
+                    ...period,
+                });
+                this.$nextTick(() => {
+                    this.close();
+                });
+            },
+            dateChange(date) {
+                // parse selected dates and convert to correct timezone
+                // the datepicker returns dates in browser timezone
+                // we should emit values in profile timezone
+                // so we have to convert
+                // the problem details: if your browser timezone is +3 then you will have .toISOString as 21:00
+                // and user profile timezone will never change that value
+                const start = this.timezoneHandler().readLocalDate(date.start.format(DateTimeFormats.datePickerDate)).startOf('day');
+                const end = this.timezoneHandler().readLocalDate(date.end.format(DateTimeFormats.datePickerDate)).endOf('day');
+                this.$emit('input', {
+                    isRelative: false,
+                    start,
+                    end,
+                });
+                this.$nextTick(() => {
+                    this.close();
+                });
+            },
+        },
+    };
+</script>
+
+<style lang="scss">
```

Comment:
Just curious, I see scoped styles here but I think that we don't "allow" them in recomm. Do you know where can I find a document explaining these kind of conventions?

Reply:
Usage of scoped styles is the only possible way to get styles in web-component mode. During `es` build all styles are extracted to separate stylesheet and after that they can be imported manually.

Anyway all styles are stored in separate scss files + we have a build `build:theme` to... build a theme 😊 So recomm could not be a reference here.

Reply:
Oh yeah, makes sense. Thanks for clarifying.